src/lm_robust_helper.cpp

```diff
@@ -195,27 +193,25 @@ List lm_robust_helper(const arma::vec & y,
           arma::mat At_WX = mat_sqrt_inv(
             I_min_H - arma::trans(H) + Xoriginal.rows(cluster_ids) * MUWTWUM * arma::trans(Xoriginal.rows(cluster_ids)),
             true
-          ) * arma::trans(Xt.cols(cluster_ids)) ;
+          ) * X.rows(cluster_ids) ;
 
-          //arma::mat A = D * arma::sqrtmat_sympd(arma::pinv(
-          //  D * (I_min_H) * D * D
-          //)) * D * X.rows(cluster_ids) ;
+          if (ci) {
+            arma::mat ME = (XtX_inv / weight_mean) * arma::trans(At_WX);
 
-          arma::mat ME = (XtX_inv / weight_mean) * arma::trans(At_WX);
+            P_diags.col(clusternum) = arma::sum(arma::pow(ME, 2), 1);
```

Comment:
arma has a norm function for calculating two norms / frobenius norms.

Reply:
Thanks for that, but this summing by row. Can it handle that?